web/yo/app/scripts/controllers/gene.js

```diff
@@ -1852,6 +1910,42 @@ angular.module('oncokbApp')
             $scope.getLastReviewedCancerTypesName = function (tumor) {
                 return mainUtils.getCancerTypesName(tumor.cancerTypes_review.lastReviewed);
             };
+
```

Comment:
removeTherapyInMap and saveToMap functions are so similar. Could you extract common part of them or merge them together? For example,
```
function example(type, ...) { 
                var indices = getIndexByPath(mapPath);
                var geneName = $scope.gene.name;
                var mutationName = $scope.gene.mutations[indices[0]].name;
                var cancerTypes = $scope.gene.mutations[indices[0]].tumors[indices[1]].cancerTypes;

                if (type === 'remove') {
                        removeTherapyInMap(keys, therapyUuid, mapPath);
                } else {  
                        saveToMap(therapyUuid, name_uuid, name, mapPath);
                } 
}
```